pysoa/common/transport/local.py

```diff
@@ -43,40 +47,61 @@ def __init__(self, service_name, server_class_or_path, server_settings):
             'path': 'pysoa.common.transport.local:ThreadlocalServerTransport',
         }
 
+        # Set an empty queued request; we'll use this later
+        self._current_request = None
+
+        # Set up a deque for responses for just this client
+        self.response_messages = deque([])
```

Comment:
In ASGI the client would start consuming the responses before the server had finished them, though. In reality we'd never have a max len any code would hit.